napari/settings/_base.py

```diff
@@ -508,18 +508,18 @@ def _remove_bad_keys(data: dict, keys: list[tuple[int | str, ...]]):
 def _restore_config_data(dct: dict, delete: dict, defaults: dict) -> dict:
     """delete nested dict keys, restore from defaults."""
     for k, v in delete.items():
-        # restore from defaults if present, or just delete the key
-        if k in dct:
-            if k in defaults:
-                dct[k] = defaults[k]
-            else:
-                del dct[k]
         # recurse
-        elif isinstance(v, dict):
-            dflt = defaults.get(k)
+        if isinstance(v, dict):
```

Comment:
Is `_restore_config_data` function called anywhere in the code? 

Your code looks fine. I'm just not seeing it used anywhere.

Reply:
It is called from `_remove_env_settings` that is called from `EventedConfigFileSettings,dict` method